### PR TITLE
[Maps - Tracking] Review request for Search to add version preview/2022-09-01-preview [Hub Generated] 

### LINKED ISSUE #20297: [Maps - Tracking] API Review
<div data-wrapper="true" style="font-size:9pt;font-family:'Segoe UI','Helvetica Neue',sans-serif;"><div>This meeting was created by Igor Shishkin.</div>
<div>&nbsp;</div>
<div>It will be used to Track the conversation in the informational Session for the Azure Maps Service.</div>
<div>&nbsp;</div>
<div>Detailed meeting information and documents provided can be accessed&nbsp;<a href="https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=c3c76051-1a1f-ed11-b83e-0022480ae914">here</a></div></div>

### PR DESCRIPTION
**Pull request by specialforest**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/specialforest/azure-rest-api-specs/dev-maps-Search-2022-09-01-preview...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:https://microsoft.sharepoint.com/:w:/t/Azure-Maps/EQaM-9d6n2FNvxvZpqUJhAgBvANwf5GksWTC5wK6wOhADA?e=tCYbNm
* Previous Open API Doc:
* Updated paths: /search/polygon

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Comment by specialforest** [comment]
This is blocked by https://github.com/Azure/azure-rest-api-specs/pull/20244

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/examples/GetPolygon.json [hunk 1]
```diff
@@ -0,0 +1,627 @@
...
+  "parameters": {
+    "api-version": "2022-09-01-preview",
+    "location": "47.612560,-122.204141",
+    "resolution": "small",
+    "entityType": "locality"
+  },
+  "responses": {
+    "200": {
+      "headers": {
+        "Content-Type": "application/geo+json"
```
Comment (stack111) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/examples/GetPolygon.json:11]: >"application/geo+json" [](http://example.com/codeflow?start=24&length=22)

@specialforest will follow up with you offline. We might want to drive consistency between revgc and autosuggest for this.

**Review by markweitzel — APPROVED** [review]
Per #20297 approved

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 2]
```diff
@@ -0,0 +1,966 @@
...
+      "x-ms-parameter-location": "client"
+    },
+    "Top": {
+      "name": "top",
+      "in": "query",
+      "description": "Maximum number of responses that will be returned. Default: 5, minimum: 1 and maximum: 20.",
+      "required": false,
+      "type": "integer",
+      "format": "int32",
+      "default": 5,
```
Comment (mikekistler) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:60]: top should not have a default or maximum.

**Review by mikekistler — APPROVED** [review]
Looks good but I think you should remove the default and maximum from the `top` parameter.

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 3]
```diff
@@ -0,0 +1,966 @@
...
+            },
+            "matchCodes": {
+              "$ref": "#/definitions/MatchCodes"
+            },
+            "address": {
+              "$ref": "#/definitions/Address"
+            },
+            "geocodePoints": {
+              "$ref": "#/definitions/GeocodePoints"
+            }
```
Comment (mikekistler) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:753]: Recommend to flatten these properties in the REST API. Our client generators have deprecated `x-ms-client-flatten` so if having the properties at the top level is best for the user experience then simply do this at the REST API level.

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 4]
```diff
@@ -0,0 +1,966 @@
...
+          "description": "ID for feature returned"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "One of: \n* Address\n* RoadBlock\n* RoadIntersection",
+              "readOnly": true
```
Comment (mikekistler) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:740]: Recommend to remove `readOnly: true` if this schema is not used in a request-body.

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 5]
```diff
@@ -0,0 +1,966 @@
...
+        "locality": {
+          "description": "locality property",
+          "type": "string",
+          "readOnly": true
+        },
+        "neighborhood": {
+          "description": "neighborhood property",
+          "type": "string",
+          "readOnly": true
+        },
```
Comment (mikekistler) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:800]: Recommend to remove `readOnly: true` if this schema is not used in a request-body.

**Review by mikekistler — COMMENTED** [review]
Found a few more minor issues I hope you can address either now or before GA.

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json (hunk 5 above)
Comment (specialforest) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json]: updated

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 6]
```diff
@@ -0,0 +1,966 @@
...
+              "description": "Number of successful requests in the batch",
+              "type": "integer",
+              "format": "int32",
+              "readOnly": true
+            },
+            "totalRequests": {
+              "description": "Total number of requests in the batch",
+              "type": "integer",
+              "format": "int32",
+              "readOnly": true
```
[the remark this answers is not in the document]
Comment (specialforest) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json]: updated

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 7]
```diff
@@ -0,0 +1,966 @@
...
+        "summary": {
+          "description": "Summary for the batch request",
+          "type": "object",
+          "readOnly": true,
+          "properties": {
+            "successfulRequests": {
+              "description": "Number of successful requests in the batch",
+              "type": "integer",
+              "format": "int32",
+              "readOnly": true
```
[the remark this answers is not in the document]
Comment (specialforest) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json]: updated

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 8]
```diff
@@ -0,0 +1,966 @@
...
+      }
+    },
+    "GeocodingBatchResponse": {
+      "description": "This object is returned from a successful Geocoding Batch service call.",
+      "type": "object",
+      "properties": {
+        "summary": {
+          "description": "Summary for the batch request",
+          "type": "object",
+          "readOnly": true,
```
[the remark this answers is not in the document]
Comment (specialforest) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json]: updated

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json (hunk 2 above)
Comment (specialforest) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json]: removed

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 9]
```diff
@@ -0,0 +1,966 @@
...
+          "description": "The name associated with the geographical area.",
+          "type": "string"
+        },
+        "copyright": {
+          "description": "The copyright string.",
+          "type": "string"
+        },
+        "copyrightURL": {
+          "description": "A URL that lists many of the data providers for Azure Maps and their related copyright information.",
+          "type": "string"
```
Comment (specialforest) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:940]: FYI I tried adding "format: uri" and "format: url", but I get an error when running validation locally that the "uri" is not known format.

**Pull request closed without merging**